### DISCUSSION TITLE
Custom element with both value and type attributes throws an unknown error - any way to handle?

### DISCUSSION THREAD

**Discussion by Jamie452** [rehypejs/rehype#151] [Q&A]
Our use case led us to use rehype to parse HTML, perform transformations, and then output back into HTML.

Rehype has worked great so far, but we've just tried inputting a custom HTML element, which has both `type` and `value` attributes and are hitting the following error.

Input:

`<my-custom-tag type="some type" value="some value"></my-custom-tag>`

Error:

`Error: Cannot compile unknown node "some type"`

I suspect the AST might rely on `type` + `value` internally, and might be making an exception for handling these when they exist on built in elements such as `input`, which works fine (`<input type="some type" value="some value"></input>`).

I'm wondering if theres a way to specify how to handle these existing on custom elements?

I saw something which I think is somewhat related on the remark-rehype package, but we're not using markdown (https://github.com/remarkjs/remark-rehype#unknown-nodes).

**Comment by ChristianMurphy** [comment]
Thanks for reaching out @Jamie452!
There may be an issue, but I don't think it is with the serializer.
Looking at https://codesandbox.io/p/sandbox/youthful-dream-znhyqc?file=%2Fsrc%2Findex.ts%3A15%2C30
The HTML AST (hast) can represent a `type` and `value` property without conflict, inside the `properties` object on the AST.
But the parser seems to be pulling the attributes in as children of the HTML element, instead of as `properties`.

It could either be a bug in the original parser https://github.com/inikulin/parse5 or in translator for parse5 to hast https://github.com/syntax-tree/hast-util-from-parse5
@Jamie452 would you be interested in digging in a bit more and tracing the root cause? Maybe opening a PR with a fix?
